Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -24,6 +24,14 @@ public interface OutputPrinter
     void WriteTrailer(PipelineStatistics stats);
     void WriteErrorInformation(ErrorInformation errorInfo, TextWriter tw, bool skipExecutionTrace = true);
     void ReportBplError(IToken tok, string message, bool error, TextWriter tw, string category = null);
+
+    void ReportImplementationsBeforeVerification(Implementation[] implementations);
+    // For real-time feedback.
+    void ReportStartVerifyImplementation(Implementation implementation);
+    void ReportEndVerifyImplementation(Implementation implementation, VerificationResult result);
+    
+    // non-positive for default priority.
+    int GetVerificationPriority(IToken implTok);
```

Comment:
This method doesn't match the name of the interface `OutputPrinter`. Why do you need this to begin with? Can't you use the `{:priority}` attribute on Implementations to control their priority?

Reply:
What do you mean by "doesn't match the name of the interface `OutputPrinter`"?

I need this method so that if a particular output printer is temporarily interested by the result of verifying a specific implementation first, it can set a priority independent of the source code, e.g. for an IDE extension where the user wants to verify the last edited implementation.
You can still use the `{:priority}` attribute on implementations to manually control their priority.

One debatable question is, how to merge a manual and an output-printer given priority. In this PR, I give priority to the output printer - in the IDE extension I'm developing, it will always verify the last five implementations. We could also max the priorities. But I found that, in an IDE setting, it makes it possible to choose the order of verification without writing explicit `{:priority}` attributes, which is pretty neat.

Reply:
I understand why you added the method, but why don't you use the priority attribute instead?

Reply:
Aahhhh ok got it got it. I historically discovered the {:priority} attribute while writing this code, but I didn't know at that time I could change the implementation, I was trying to only modify the verification behavior. But now you suggestion makes sense to me. I implemented in my branch something where it sets priorities of implementations in the newly added callback, so that there is no need to order them later.
Everything is working ! thanks for the suggestion.